packages/amplify-category-api/src/index.ts

```diff
@@ -205,3 +215,172 @@ export async function handleAmplifyEvent(context, args) {
   context.print.info(`${category} handleAmplifyEvent to be implemented`);
   context.print.info(`Received event args ${args}`);
 }
+
+type ApiResource = {
+  category: string;
+  resourceName: string;
+  githubInfo: {
+    path: string;
+    tokenSecretArn: string;
+  };
+  deploymentMechanism: DEPLOYMENT_MECHANISM;
+  authName: string;
+  restrictAccess: boolean;
+  lastPushTimeStamp: string;
+  dependsOn: ResourceDependency[];
+  environmentMap: Record<string, string>;
+  categoryPolicies: any[];
+  mutableParametersState: any;
+  output?: Record<string, any>;
+};
+
+export async function generateContainersArtifacts(context: any, resource: ApiResource) {
```

Comment:
nit: would rather place this function in utils and just export it here in index.ts rather than populating all the logic in here.